src/pages/index.js

```diff
@@ -0,0 +1,329 @@
+const previewModalCloseButton = previewModal.querySelector(".modal__close-button_preview");
+
+//Avatar Form Elements
+const avatarModal = document.getElementById("avatar-modal");
+const avatarFormElement = avatarModal.querySelector(".modal__form");
+const avatarSubmitButton = avatarModal.querySelector(".modal__submit-button");
+const avatarModalCloseButton = avatarModal.querySelector(".modal__close-button");
+
+//Delete Card Elements
+const cardDeleteModal = document.querySelector("#delete-modal");
+const cardDeleteForm = cardDeleteModal.querySelector(".modal__form");
+
+
+let selectedCard, selectedCardId;
+
+
+function getCardElement(data) {
+    const cardElement = cardTemplate.content.querySelector(".card").cloneNode(true);
+    
+    cardElement.setAttribute("data-card-id", data._id);
+
+    const cardNameElement = cardElement.querySelector(".card__title");
+    const cardImageElement = cardElement.querySelector(".card__image");
+    const cardLikeButton = cardElement.querySelector(".card__like-button");
+    const cardDeleteButton = cardElement.querySelector(".card__delete-button");
+
+    if (data.isLiked) {
+      cardLikeButton.classList.add("card__like-button_liked");
+    }
+  
+    cardImageElement.setAttribute("src", data.link);
+    cardImageElement.setAttribute("alt", data.name);
+    cardNameElement.textContent = data.name;
+    
+    cardImageElement.addEventListener("click", () => {
+        openModal(previewModal);
+
+        previewModalImageElement.src = data.link;
+        previewModalImageElement.alt = data.name;
+        previewModalCaptionElement.textContent = data.name;
+    });
+
+    cardLikeButton.addEventListener("click", (evt) => handleLikeCard(evt, data._id, data.isLiked));
+
+    cardDeleteButton.addEventListener("click", () => handleDeleteCard(cardElement, data._id));
+
+    return cardElement;
+
+}
+
+
+function openModal (modal) {
+    modal.classList.add("modal_opened");
+    document.addEventListener("keydown", handleEscape);
+}
+
+function closeModal (modal) {
+    modal.classList.remove("modal_opened");
+    document.removeEventListener("keydown", handleEscape);
+}
+
+function handleLikeCard(event, cardId, isLiked) {
+  api.updateLikeCard(cardId, isLiked)
+    .then(() => {
+      event.target.classList.toggle("card__like-button_liked");
+    })
+    .catch(console.error);
+}
+
+function handleProfileFormSubmit(evt) {
+    evt.preventDefault();
+
+    const submitButton = evt.submitter;
+    setButtonText(submitButton, true, "Saving...");
+
+    api.updateUserInfo({name: editProfileNameInput.value, about: editProfileDescriptionElement.value})
+    .then((userData) => {
+        profileNameElement.textContent = userData.name;
+        profileDescriptionElement.textContent = userData.about;
+        closeModal(editProfileModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitButton, true, "Saving...", "Save");
```

Comment:
`true` 

In `.finally`, you need to pass `false` to stop the loading effect


And `"Saving...", "Save"` are not needed because you have them by default inside `setButtonText`


```js

      setButtonText(submitButton, false);

```

**Please, correct it everywhere in the code**